x-pack/plugins/canvas/public/lib/aeroelastic/index.d.ts

```diff
@@ -10,21 +10,20 @@
 type f64 = number; // eventual AssemblyScript compatibility; doesn't hurt with vanilla TS either
 type f = f64; // shorthand
 
-export type vector2d = [f, f, f] & ReadonlyArray<f> & { __nominal: 'vector2d' };
-export type vector3d = [f, f, f, f] & ReadonlyArray<f> & { __nominal: 'vector3d' };
+export type Vector2d = Readonly<[f, f, f]>;
+export type Vector3d = Readonly<[f, f, f, f]>;
 
-export type transformMatrix2d = [f, f, f, f, f, f, f, f, f] &
-  ReadonlyArray<f> & { __nominal: 'transformMatrix2d' };
-export type transformMatrix3d = [f, f, f, f, f, f, f, f, f, f, f, f, f, f, f, f] &
-  ReadonlyArray<f> & { __nominal: 'transformMatrix3d' };
+export type Matrix2d = [f, f, f, f, f, f, f, f, f];
```

Comment:
Why do we need to export `Matrix2d` and `Matrix3d`, couldn't we use `TransformMatrix2d` and `TransformMatrix3d` in `matrix.ts` and `matrix2d.ts` respectively? If not, interested to learn

Reply:
Note where `Matrix3d` is used: `matrix.ts` line 174.

The `as` usage was masking the error in 201: the `const` on 174 is being manipulated, illegal for a `ReadOnly` array.

I exported `Matrix2d` for completeness

Reply:
Indeed, good point!